foundations/src/telemetry/server.rs

```diff
@@ -135,6 +135,7 @@ fn bind_socket(addr: SocketAddr) -> BootstrapResult<Socket> {
     )?;
 
     socket.set_reuse_address(true)?;
+    #[cfg(all(unix, not(any(target_os = "solaris", target_os = "illumos"))))]
```

Comment:
I think both "solaris" and "illumos" are very exotic build targets nowadays, so we can ignore them and just simplify it to `#[cfg(unix)]`

Reply:
Alright! I copied that straight from the source of socket2. I'll update that.